django/core/management/commands/makemessages.py

```diff
@@ -382,7 +382,11 @@ def handle(self, *args, **options):
             potfiles = self.build_potfiles()
 
             # Build po files for each selected locale
+            valid_locale = re.compile(r'^[a-z]{1,8}((_{1})([A-Z]{1})'
+                                      r'([a-z]{1,8}|[A-Z]{1,2}))?(_([A-Z]{1,4}|[0-9]{2,4}))?$')
             for locale in locales:
+                if not valid_locale.match(locale):
+                    self.stdout.write("invalid locale %s" % locale)
                 if self.verbosity > 0:
```

Comment:
Need to add `continue` before exiting your new branch, otherwise it will be misleading for the command to then print `processing locale %s`.
This means you'll need to update your test, also.

Reply:
Single quotes, not `""`.

Reply:
@pope1ni in [this comment](https://code.djangoproject.com/ticket/29712#comment:15) only mentions skipping the `processing locale %s` statement, not sheer `continue`ing, so maybe to be safer: you could just make the next branch `elif self.verbosity > 0:` for now. Warning users when things silently fail is the point, not preventing users from using yet-to-be-invented locale codes.

Reply:
IMO a warning is to be printed even when the verbosity is equaled to 0 as it is a statement that the locale is invalid and this won't depend on verbosity level. Shall I change my branch to something like this:
**From :** 
`if not valid_locale.match(locale):`
`                    self.stdout.write("invalid locale %s" % locale)`
`                if self.verbosity > 0:`
`                    self.stdout.write('processing locale %s' % locale)`
**To:**
`if not valid_locale.match(locale) and self.verbosity > 0:`
`                    self.stdout.write("invalid locale %s" % locale)`
`                elif self.verbosity > 0:`
`                    self.stdout.write('processing locale %s' % locale)`

Reply:
As the makemessage will make a directory for locale and will process even the invalid locales. So IMO, when the verbosity is > 0 it should print `processing locale` as the command is processing the same.

Reply:
I agree we don't need to check verbosity for the warning you are adding, but I wasn't suggesting that. I was just questioning whether we should be printing `"processing..."` when we're warning about a potential silent failure. Sort of contradicts the warning (is it invalid or is it being processed?)

The code you proposed sounds more like what I described than you are describing, but it's hard to tell without block-quoting to see the indentation. Use three backticks to
```
quote
   several
        lines
```

Reply:
Ah I see, you've already pushed the change I suggested. LGTM. We should wait to hear advice from more experienced contributors in this area. 👍

Reply:
Yeah I think the same.